app/main.py

```diff
@@ -1,2 +1,99 @@
 class Dictionary:
-    pass
+
+    INITIAL_SIZE = 8
+    LOAD_FACTOR = 2 / 3
+    RESIZE_FACTOR = 2
+
+    def __init__(self, elements=None):
+        self.size = self.INITIAL_SIZE
+        self.buckets = [[] for _ in range(self.size)]
+        if elements:
+            for key, value in elements:
+                self.__setitem__(key, value)
+
+    def __setitem__(self, key, value):
+        hashed_value = hash(key)
+        bucket = self.buckets[hashed_value % self.size]
+
+        # rewrite the value of existing key:
+        for ind, element in enumerate(bucket):
+            if element[0] == key:
+                bucket[ind] = (key, hashed_value, value)
+                break
+
+        # create new key-value pair and check for the need to resize
+        if (key, hashed_value, value) not in bucket:
+            fullness = (sum(1 for el in self.buckets if el) + 1) / self.size
```

Comment:
not the best way to calculate, you make this long math, try to find a faster way